File: Java/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int open = 0, closed = 0;

		for (var ch : s.toCharArray()) {
            if (ch == '(') {
                open++;
            }
            else {
                if (open <= 0) {
                    closed++;
                }
                else {
                    open--;
                }
            }
        }

        return open + closed;
    }
}

File: Python/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        open, extra = 0, 0
        for ch in s :
            if ch == '(' :
                open += 1
            else :
                if open <= 0 :
                    extra += 1
                else :
                    open -= 1
        return open + extra